File: dart_sample/lib/counter_scope.dart
import 'package:flutter/widgets.dart';

import 'counter_app_model.dart';

class CounterScope extends InheritedNotifier<CounterAppModel> {
  const CounterScope({
    super.key,
    required CounterAppModel super.notifier,
    required super.child,
  });

  static CounterAppModel of(BuildContext context) {
    final scope = context.dependOnInheritedWidgetOfExactType<CounterScope>();
    if (scope == null || scope.notifier == null) {
      throw StateError('CounterScope not found in widget tree.');
    }

    return scope.notifier!;
  }
}

File: src/Sample/Flutter.Net/CounterScope.cs
using System;
using Flutter.Widgets;

namespace Flutter.Net;

public sealed class CounterScope : InheritedNotifier<CounterAppModel>
{
    public CounterScope(CounterAppModel notifier, Widget child) : base(notifier, child)
    {
    }

    public static CounterAppModel Of(BuildContext context)
    {
        var scope = context.DependOnInherited<CounterScope>()
                    ?? throw new InvalidOperationException("CounterScope not found in widget tree.");
        return scope.Notifier ?? throw new InvalidOperationException("CounterScope notifier is null.");
    }
}
